flo-runner/src/test/java/com/spotify/flo/context/OverridingContextTest.java

```diff
@@ -60,11 +60,13 @@ public String provide(EvalContext evalContext) {
     }
   };
 
-  int countUpstreamRuns = 0;
-  int countRootRuns = 0;
+  private static int countUpstreamRuns = 0;
```

Comment:
It's kind of subtle to understand why they have to be `static` unfortunately. Might be worth commenting a bit.